### ISSUE TITLE
Should reject overriding variadic param in a descendant with array param

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#1422]
https://psalm.dev/r/c6c255ff92
https://3v4l.org/m7eRM
```php
<?php
  interface I {
    public function f(float ...$rest): void;
  }

  class C implements I {
    /** @param array<int,float> $f */
    public function f($f): void {}
  }
```
**Expected**: MethodSignatureMismatch, as it's certain fatal error
**Actual**:
```
Psalm output (using commit 8ec188a): 

No issues!
```
```
Output for 7.1.25 - 7.3.2
Fatal error: Declaration of C::f($f): void must be compatible with I::f(float ...$rest): void in /in/m7eRM on line 6

Process exited with code 255.
```

**Comment by muglug** [comment]
Was fixed with the fix to #1421

**Comment by weirdan** [comment]
Another variant (with default added) was not:

https://3v4l.org/4lqKf
https://psalm.dev/r/2dce878542
```php
<?php
  interface I {
    public function f(float ...$rest): void;
  }

  class C implements I {
    /** @param array<int,float> $f */
    public function f($f = []): void {}
  }
```
```
Psalm output (using commit f9b1a63): 

No issues!
```
```
Output for 7.1.25 - 7.3.2
Fatal error: Declaration of C::f($f = Array): void must be compatible with I::f(float ...$rest): void in /in/4lqKf on line 6

Process exited with code 255.
```
